sources/api/apiclient/src/network.rs

```diff
@@ -0,0 +1,262 @@
+//! This module implements network configuration API calls.
+//! It supports sourcing `net.toml` configuration files from the filesystem or base64 encoded strings.
+
+use base64::{engine, Engine};
+use snafu::{OptionExt, ResultExt};
+use std::path::Path;
+use tokio::io::AsyncReadExt;
+use url::Url;
+
+/// Configures network settings by sending the provided content to the API server.
+///
+/// The configuration will be applied at the next boot - a reboot is required for changes to take effect.
+pub async fn configure<P>(socket_path: P, content: String) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let uri = "/actions/network/configure";
+    let method = "POST";
+    let (_status, _body) = crate::raw_request(&socket_path, uri, method, Some(content))
+        .await
+        .context(error::ConfigureRequestSnafu { uri, method })?;
+
+    Ok(())
+}
+
+/// Retrieves the network configuration content from the given source URI.
+///
+/// Supports file:// and base64: URI schemes. If no input source is provided, reads from stdin.
+pub async fn get_content<S>(input_source: Option<S>) -> Result<String>
+where
+    S: Into<String>,
+{
+    get_content_with_reader(input_source, tokio::io::stdin()).await
```

Comment:
It feels somewhat like `get_content` should be deciding whether or not to use the `input_source` or `stdin` rather than passing both to the delegate function and allowing it to decide.